rpcs3/Emu/Memory/Memory.cpp

```diff
@@ -13,136 +13,94 @@ VirtualMemoryBlock* VirtualMemoryBlock::SetRange(const u32 start, const u32 size
-		}
-	}
-
 	m_mapped_memory.emplace_back(addr, realaddr, size);
 	return true;
 }
 
-bool VirtualMemoryBlock::UnmapRealAddress(u32 realaddr, u32& size)
+u32 VirtualMemoryBlock::UnmapRealAddress(u32 realaddr)
 {
 	for (u32 i = 0; i<m_mapped_memory.size(); ++i)
 	{
 		if (m_mapped_memory[i].realAddress == realaddr && IsInMyRange(m_mapped_memory[i].addr, m_mapped_memory[i].size))
 		{
-			size = m_mapped_memory[i].size;
+			const u32 size = m_mapped_memory[i].size;
 			m_mapped_memory.erase(m_mapped_memory.begin() + i);
-			return true;
+			return size;
 		}
 	}
 
-	return false;
+	return 0;
 }
 
-bool VirtualMemoryBlock::UnmapAddress(u32 addr, u32& size)
+u32 VirtualMemoryBlock::UnmapAddress(u32 addr)
 {
 	for (u32 i = 0; i<m_mapped_memory.size(); ++i)
 	{
 		if (m_mapped_memory[i].addr == addr && IsInMyRange(m_mapped_memory[i].addr, m_mapped_memory[i].size))
 		{
-			size = m_mapped_memory[i].size;
+			const u32 size = m_mapped_memory[i].size;
 			m_mapped_memory.erase(m_mapped_memory.begin() + i);
-			return true;
+			return size;
 		}
 	}
 
-	return false;
+	return 0;
 }
 
-bool VirtualMemoryBlock::Read32(const u32 addr, u32* value)
+s64 VirtualMemoryBlock::Read32(const u32 addr)
 {
-	u32 realAddr;
-	if (!getRealAddr(addr, realAddr))
-		return false;
-	*value = vm::read32(realAddr);
-	return true;
+	if (const u32 realAddr = getRealAddr(addr))
+	{
+		return static_cast<u64>(vm::read32(realAddr));
+	}
+
+	return -1;
 }
 
 bool VirtualMemoryBlock::Write32(const u32 addr, const u32 value)
 {
-	u32 realAddr;
-	if (!getRealAddr(addr, realAddr))
-		return false;
-	vm::write32(realAddr, value);
-	return true;
+	if (const u32 realAddr = getRealAddr(addr))
+	{
+		vm::write32(realAddr, value);
+		return true;
+	}
+	return false;
 }
 
-bool VirtualMemoryBlock::getRealAddr(u32 addr, u32& result)
+u32 VirtualMemoryBlock::getRealAddr(u32 addr)
 {
-	for (u32 i = 0; i<m_mapped_memory.size(); ++i)
+	for (const auto mem_block : m_mapped_memory)
 	{
-		if (addr >= m_mapped_memory[i].addr && addr < m_mapped_memory[i].addr + m_mapped_memory[i].size)
+		if (addr >= mem_block.addr && addr < mem_block.addr + mem_block.size)
 		{
-			result = m_mapped_memory[i].realAddress + (addr - m_mapped_memory[i].addr);
-			return true;
+			return mem_block.realAddress + (addr - mem_block.addr);
 		}
 	}
 
-	return false;
+	return 0;
 }
 
 u32 VirtualMemoryBlock::getMappedAddress(u32 realAddress)
 {
-	for (u32 i = 0; i<m_mapped_memory.size(); ++i)
+	for (const auto mem_block : m_mapped_memory)
 	{
-		if (realAddress >= m_mapped_memory[i].realAddress && realAddress < m_mapped_memory[i].realAddress + m_mapped_memory[i].size)
+		if (realAddress >= mem_block.realAddress && realAddress < mem_block.realAddress + mem_block.size)
 		{
-			return m_mapped_memory[i].addr + (realAddress - m_mapped_memory[i].realAddress);
+			return mem_block.addr + (realAddress - mem_block.realAddress);
 		}
 	}
 
-	return 0;
+	return -1u;
```

Comment:
Slight nitpick but half these either return 0 or return -1 now? maybe standardize or maybe std::optional?

Reply:
this was a bug, io address 0 is valid if you remember the testcase